Metal does not load any more without error when installation not functional

```julia
 ERROR: LoadError: The following 1 direct dependency failed to precompile:

BFloat16sExt --code-coverage=@/home/runner/work/CellArrays.jl/CellArrays.jl --color=yes --check-bounds=yes --warn-overwrite=yes --depwarn=yes --inline=yes --startup-file=no --track-allocation=none --check-bounds=yes --compiled-modules=yes 

Failed to precompile BFloat16sExt [30729c67-1fb4-564a-af8f-be693f25f210] to "/home/runner/.julia/compiled/v1.11/BFloat16sExt/jl_gVWBL2".
ERROR: LoadError: could not load symbol "sysctlbyname":
/opt/hostedtoolcache/julia/1.11.0/x64/bin/julia: undefined symbol: sysctlbyname
Stacktrace:
 [1] _syscall_version(name::String)
   @ Metal ~/.julia/packages/Metal/hYqUg/src/version.jl:5
 [2] macos_version()
   @ Metal ~/.julia/packages/Metal/hYqUg/src/version.jl:43
 [3] top-level scope
   @ ~/.julia/packages/Metal/hYqUg/ext/BFloat16sExt.jl:7
 [4] include
   @ ./Base.jl:557 [inlined]
 [5] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt128}}, source::Nothing)
   @ Base ./loading.jl:2806
 [6] top-level scope
   @ stdin:4
in expression starting at /home/runner/.julia/packages/Metal/hYqUg/ext/BFloat16sExt.jl:1
in expression starting at stdin:
in expression starting at /home/runner/work/_actions/julia-actions/julia-runtest/v1/test_harness.jl:9
Error: Process completed with exit code 1.
```
this is from the CI of [CellArrays.jl](https://github.com/omlins/CellArrays.jl) : 
https://github.com/omlins/CellArrays.jl/actions/runs/11271443359/job/31344456428?pr=40#step:6:285

@maleadt This is a similar issue as: https://github.com/JuliaGPU/AMDGPU.jl/issues/685

--- a/ext/BFloat16sExt.jl
+++ b/ext/BFloat16sExt.jl
@@ -4,9 +4,11 @@ using Metal: MPS.MPSDataType, MPS.MPSDataTypeBFloat16, MPS.jl_mps_to_typ, macos_
 using BFloat16s
 
 # BFloat is only supported in MPS starting in MacOS 14
-if macos_version() >= v"14"
-    Base.convert(::Type{MPSDataType}, ::Type{BFloat16}) = MPSDataTypeBFloat16
-    jl_mps_to_typ[MPSDataTypeBFloat16] = BFloat16
+@static if Sys.isapple()
+    if macos_version() >= v"14"
+        Base.convert(::Type{MPSDataType}, ::Type{BFloat16}) = MPSDataTypeBFloat16
+        jl_mps_to_typ[MPSDataTypeBFloat16] = BFloat16
+    end
 end
 
 end # module

